proxy/envoy/config.go

```diff
@@ -126,9 +126,47 @@ func Generate(instances []*model.ServiceInstance, services []*model.Service, rul
 // buildListeners uses iptables port redirect to route traffic either into the
 // pod or outside the pod to service clusters based on the traffic metadata.
 func buildListeners(instances []*model.ServiceInstance,
-	services []*model.Service,
+	services []*model.Service, rules []*config.RouteRule,
+	upstreams []*config.UpstreamCluster,
```

Comment:
Upstream makes sense in the context of an individual rule in a proxy. If you have multiple federated clusters and also ingress controllers like we plan to have in Manager, upstream cluster is just confusing.